From ed549035a57f287a62d3ea4d21d48b851701f8ca Mon Sep 17 00:00:00 2001
From: Wei-Chiu Chuang <weichiu@apache.org>
Date: Fri, 9 Dec 2016 16:07:56 -0800
Subject: [PATCH 2111/2848] HDFS-11229. HDFS-11056 failed to close meta file.
 Contributed by Wei-Chiu Chuang.

(cherry picked from commit 292bd78b4494ba1a1073b32b3f89d95e72f3204d)
(cherry picked from commit fa3b2bee618c72bae57890f011b475522fe2ff47)

Change-Id: Ib44ad4fd1e3ffa4271d17f1791a31d3852569f0c
---
 .../datanode/fsdataset/impl/FsDatasetImpl.java     |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java
index d4995ce..41d6f7c 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java
@@ -1148,9 +1148,10 @@ public synchronized ReplicaHandler append(ExtendedBlock b,
     int offsetInChecksum = BlockMetadataHeader.getHeaderSize() +
         (int)(onDiskLen / bytesPerChecksum * checksumSize);
     byte[] lastChecksum = new byte[checksumSize];
-    RandomAccessFile raf = new RandomAccessFile(metaFile, "r");
-    raf.seek(offsetInChecksum);
-    raf.read(lastChecksum, 0, checksumSize);
+    try (RandomAccessFile raf = new RandomAccessFile(metaFile, "r")) {
+      raf.seek(offsetInChecksum);
+      raf.read(lastChecksum, 0, checksumSize);
+    }
     return lastChecksum;
   }
 
-- 
1.7.9.5

